presto-hive/src/main/java/io/prestosql/plugin/hive/util/InternalHiveSplitFactory.java

```diff
@@ -193,14 +193,28 @@ public String getPartitionName()
                 s3SelectPushdownEnabled && S3SelectPushdown.isCompressionCodecSupported(inputFormat, path)));
     }
 
-    private static void checkBlocks(List<InternalHiveBlock> blocks, long start, long length)
+    private static void checkBlocks(Path path, List<InternalHiveBlock> blocks, long start, long length)
     {
-        checkArgument(length >= 0);
-        checkArgument(!blocks.isEmpty());
-        checkArgument(start == blocks.get(0).getStart());
-        checkArgument(start + length == blocks.get(blocks.size() - 1).getEnd());
+        checkArgument(length >= 0, "Split size is negative for %s, split from %s with length %s", path, start, length);
```

Comment:
I think this is overly verbose, while the error report is not going to be enduser-consumable anyway.

```suggestion
        checkArgument(length >= 0, "Split size is negative for %s, split start=%s,  length=%s", path, start, length);
```